clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/IamRoleCachingAgent.java

```diff
@@ -83,8 +83,13 @@ public IamRoleCachingAgent(
 
   @Override
   public CacheResult loadData(ProviderCache providerCache) {
-    AmazonIdentityManagement iam =
-        amazonClientProvider.getIam(account, Regions.DEFAULT_REGION.getName(), false);
+    // use a region from the account for correct endpoint resolution
+    String region =
+        !account.getRegions().isEmpty()
+            ? account.getRegions().get(0).getName()
```

Comment:
Are we assuming that there's only 1 region? Do we only want the first region name?

Reply:
IAM resources are global on the account, so each regional call should return the same data.